### ISSUE TITLE
PHP Notice when trying to compile a proto file and failure when creating output PHP file

### ISSUE THREAD

**Issue by bieli** [allegro/php-protobuf#10]
I get the following "PHP Notice" when trying to compile a proto file ( from Google tutorial https://developers.google.com/protocol-buffers/docs/pythontutorial). I can't see output PHP file from parser after parsing addressbook.proto.

Below I publish my steps and error message from PHP:
```
user@bieli:php-protobuf/example$ cat > addressbook.proto
package tutorial;

message Person {
  required string name = 1;
  required int32 id = 2;
  optional string email = 3;

  enum PhoneType {
    MOBILE = 0;
    HOME = 1;
    WORK = 2;
  }

  message PhoneNumber {
    required string number = 1;
    optional PhoneType type = 2 [default = HOME];
  }

  repeated PhoneNumber phone = 4;
}

message AddressBook {
  repeated Person person = 1;
}

^Z
[2]+  Stopped                 cat > addressbook.proto
user@bieli:php-protobuf/example$ php ../protoc-php.php addressbook.proto 
PHP Notice:  Undefined index: PhoneType in php-protobuf/ProtobufCompiler/ProtobufParser.php on line 1039
Type PhoneType not defined

```
*My general problem it's failure output PHP file.*

For comparision, when I using Google tools for Python ( https://developers.google.com/protocol-buffers/docs/pythontutorial ) I can see output file, after default 'protoc' command:
```
user@bieli:php-protobuf$ protoc --proto_path=./ --python_out=/tmp/ example/addressbook.proto

user@bieli:php-protobuf$ file /tmp/example/addressbook_pb2.py 
/tmp/example/addressbook_pb2.py: Python script, ASCII text executable, with very long lines
```

In Python code I can see important section:
```
_PERSON_PHONETYPE = descriptor.EnumDescriptor(
  name='PhoneType',
  full_name='tutorial.Person.PhoneType',
  filename=None,
  file=DESCRIPTOR,
  values=[
    descriptor.EnumValueDescriptor(
      name='MOBILE', index=0, number=0,
      options=None,
      type=None),
    descriptor.EnumValueDescriptor(
      name='HOME', index=1, number=1,
      options=None,
      type=None),
    descriptor.EnumValueDescriptor(
      name='WORK', index=2, number=2,
      options=None,
      type=None),
  ],
  containing_type=None,
  options=None,
  serialized_start=215,
  serialized_end=258,
)
```

I know workaround for this problem by extracting enum from message:

```
package tutorial;


enum PhoneType {
  MOBILE = 0;
  HOME = 1;
  WORK = 2;
}

message Person {
  required string name = 1;
  required int32 id = 2;
  optional string email = 3;

  message PhoneNumber {
    required string number = 1;
    optional PhoneType type = 2 [default = HOME];
  }

  repeated PhoneNumber phone = 4;
}

message AddressBook {
  repeated Person person = 1;
}
```
but maybe my issue is important for refactoring *.proto parser in PHP.

*Good luck*

**Comment by PunitSaharan** [comment]
Hi please help me i want to install php-protobuf in xamp on window.but i have no idea how it will be installed .when i run protoc-php.php its show "requires protobuf extension installed to run". how can i install this extension ??

Thanks

**Comment by lovelock** [comment]
@PunitSaharan Then your problem has nothing to do with protobuf, but you do not know how to compile a PHP extension.

```
cd php-protobuf-master
phpize
./configure
make
sudo make install
```

vim /path/to/your/php.ini
add one line

   extension=protobuf.so

**Comment by hjagodzinski** [comment]
Fixed in #74.
